unit_test/solver/network/test_network.py

```diff
@@ -0,0 +1,268 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test network class."""
+import unittest
+import uuid
+
+from simulator_core.solver.network.network import Network
+from simulator_core.solver.network.assets.solver_pipe import SolverPipe
+from simulator_core.solver.network.assets.node import Node
+
+
+class NetworkTest(unittest.TestCase):
+    """Class to test network class."""
+
+    def setUp(self) -> None:
+        """Set up for the tests."""
+        self.network = Network()
+        self.asset = SolverPipe(name=uuid.uuid4())
+        self.node = Node(name=uuid.uuid4())
+
+    def test_init(self) -> None:
+        """Test the constructor of the network class."""
+        # arrange
+
+        # act
+        network = Network()  # act
+
+        # assert
+        self.assertEqual(network.assets, {})
+        self.assertEqual(network.nodes, {})
+
+    def test_add_asset_name(self) -> None:
+        """Test adding an asset with a name."""
+        # arrange
+        asset = "Pipe"
+        asset_name = uuid.uuid4()
+
+        # act
+        name = self.network.add_asset(asset_type=asset, name=asset_name)  # act
+
+        # assert
+        self.assertEqual(name, asset_name)
+        self.assertIsInstance(self.network.assets[name], SolverPipe)
+
+    def test_add_asset_no_name(self) -> None:
```

Comment:
Zou je hier ook niet het volgende willen deon:


```suggestion
    @patch('uuid.uuid4)
    def test_add_asset_no_name(self) -> None:
        mock_uuid4.return_value = '1234'
```